src/Plug/Widget.h

```diff
@@ -0,0 +1,124 @@
+/**
+ * @file Widget.h
+ * @author Kudryashov Ilya (kudriashov.it@phystech.edu)
+ * @brief Widget interface
+ * @version 0.1
+ * @date 2023-11-17
+ *
+ * @copyright Copyright (c) 2023
+ *
+ */
+
+#ifndef __PLUG_WIDGET_H
+#define __PLUG_WIDGET_H
+
+#include "Plug/Event.h"
+#include "Plug/Graphics.h"
+#include "Plug/LayoutBox.h"
+#include "Plug/Math.h"
+
+namespace plug
+{
+
+/**
+ * @brief Interactive object with visual representation
+ *
+ */
+class Widget
+{
+public:
+  virtual ~Widget() = 0;
+
+  /**
+   * @brief Draw widget
+   *
+   * @param stack
+   * @param target buffer which the widget will be drawn on
+   */
+  virtual void draw(TransformStack& stack, RenderTarget& target) {}
```

Comment:
Why keep useless default implementation and search for bugs later? If you forget to override it, you'll never find why your widget isn't rendering properly. But if this function is pure virtual, erroneous code won't even compile

Reply:
The same can be said about event handling with absolutely the same explanation.
Not every widget needs to draw itself, just like not every widget has to handle keyboard events.
And to keep that logic transparent, we define draw method with empty implementation.

Reply:
No, for event handlers the rational default action is to do nothing. The same cannot be said about `draw()` though. Every widget has to draw itself (it is GUI element after all), so there is no good default implementation for `draw()`